json-schema-form.d.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Shorthand to lookup for keys with `x-jsf-*` preffix.
+ */
+export function pickXKey(node: Object, key: 'presentation' | 'errorMessage'): Object | undefined;
+
+type JSFConfig = {
+  initialValues?: Record<string, unknown>;
+  strictInputType?: boolean;
+  customProperties?: {
+    description: (description: string, field: $TSFixMe) => string | string;
+    [key: string]: unknown;
+  };
+  inputTypes?: {
+    errorMessage: Record<string, unknown>;
+  };
+};
+
+type Fields = Record<string, unknown>[]; //TODO: Type the field based on the given JSON Schema properties.
+
+type $TSFixMe = any;
+
+/**
+ * Returns the Yup schema structure of given fields.
+ * These fields must be the same from
+ * const { fields } = createHeadlessForm()
+ */
+export function buildCompleteYupSchema(fields: Fields, config: JSFConfig): $TSFixMe; //TODO: We need to update Yup to 1.0 which supports TS.
+
+type HeadlessFormOutput = {
+  fields: Fields;
+  handleValidation: () => $TSFixMe;
```

Comment:
docs: More docs/example for these ones:

<details>

```js
type HeadlessFormOutput = {
  /**
   * List of Fields. Each Field corresponds to a form input from the json schema.
   * @example
   * [
      {
        name: "has_pet",
        label: "Has Pet",
        options: [
          { label: "Yes", value: "yes" },
          { label: "No", value: "no" }
        ],
        required: true,
        inputType: "radio",
        jsonType: "string",
        description: "Do you have a pet?",
        errorMessage: {},
        isVisible: true
      },
      {
        type: "text",
        label: "Pet's name",
        description: "What's your pet's name?",
        required: false,
        inputType: "text",
        jsonType: "string",
        isVisible: false
      }
   * ]
   */
  fields: Fields;
  /**
   * Validates given values and returns the respective errors.
   * 
   * @example
   * const { formErrors } = handleValidation({ has_pet: "yes" });
   * console.log(formErrors) // { pet_name: "Required field." } 
  */
  handleValidation: () => $TSFixMe;
  isError: boolean;
  error?: Error;
};
```

</details>

Reply:
@gabrielseco maybe you missed this comment?

Reply:
Done!